kong/plugins/proxy-cache/handler.lua

```diff
@@ -173,11 +176,28 @@ local function cacheable_response(conf, cc)
       return false
     end
 
+    local t, subtype, params = parse_mime_type(content_type)
     local content_match = false
     for i = 1, #conf.content_type do
-      if conf.content_type[i] == content_type then
-        content_match = true
-        break
+      local expected_ct = conf.content_type[i]
+      local exp_type, exp_subtype, exp_params = parse_mime_type(expected_ct)
+      if exp_type then
+        if (exp_type == "*" or (t and lower(t) == lower(exp_type))) and
+          (exp_subtype == "*" or (subtype and
+            lower(subtype) == lower(exp_subtype))) then
+          local params_match = true
+          for key, value in pairs(exp_params or EMPTY) do
+            if value ~= (params or EMPTY)[key] then
+              params_match = false
+              break
+            end
+          end
+          if params_match and
+            (nkeys(params or EMPTY) == nkeys(exp_params or EMPTY)) then
+            content_match = true
+            break
+          end
+        end
```

Comment:
Suggesting 3 refactors here:
* parse_mime_type always will return a lowercase type/subtype (after the changes I made), we can remove the calls to `lower` from this logic
* parse_mime_type always returns a params table - sometimes it is just empty. So we can replace the `(params or EMPTY)` with `params`
* simplified "`params_match` and `content_match` + `break`" logic by `return false`

```suggestion
      local expected_ct = conf.content_type[i]
      local exp_type, exp_subtype, exp_params = parse_mime_type(expected_ct)
      if exp_type then
        if (exp_type == "*" or (t and t == exp_type)) and
          (exp_subtype == "*" or (subtype and subtype == exp_subtype)) then
          for key, value in pairs(exp_params) do
            if value ~= params[key] then
              return false
            end
          end
          if nkeys(params) ~= nkeys(exp_params) then
            return false
          end
        end
      end
```

Reply:
I was wrong at least in two accounts:
* the content type can be completely bogus, and in that case the parser will return nil on all 3 return values. Still, if one of the expected types is `*`, that is "compatible" with a bogus content type. So we need at least one `or EMPTY` for `params`
* replacing content_match, params_match and break with returns it trickier than I anticipated (it would require at least 2 extra aux functions). So I will unsay my words.